JoyboyCommunity/src/screens/Tips/index.tsx

```diff
@@ -29,77 +30,55 @@ export const Tips: React.FC = () => {
 
   const account = useAccount();
   const chainId = useChainId();
-  const sendTransaction = useTransaction();
+  const claim = useClaim();
+  const estimateClaim = useEstimateClaim();
   const walletModal = useWalletModal();
   const waitConnection = useWaitConnection();
+  const {show: showTransactionModal} = useTransactionModal();
   const {showToast} = useToast();
 
   const onClaimPress = async (depositId: number) => {
     if (!account.address) {
       walletModal.show();
-
-      const result = await waitConnection();
-      if (!result) return;
     }
 
-    const kind = NDKKind.Text;
-    const content = cairo.felt(depositId);
-    const tags = [];
-    const createdAt = Date.now();
-
-    const event = new NDKEvent(ndk);
-    event.kind = kind;
-    event.content = `claim ${content}`;
-    event.tags = tags;
-    event.created_at = createdAt;
-
-    const signature = await event.sign();
-    const signatureR = `0x${signature.slice(0, signature.length / 2)}`;
-    const signatureS = `0x${signature.slice(signature.length / 2)}`;
-
-    const publicKey = event.pubkey;
-
-    const claimCallData = CallData.compile([
-      uint256.bnToUint256(`0x${publicKey}`), // public_key
-      createdAt, // created_at
-      kind, // kind
-      byteArray.byteArrayFromString(JSON.stringify(tags)), // tags
-      content, // content
-      {
-        r: uint256.bnToUint256(signatureR), // signature R
-        s: uint256.bnToUint256(signatureS), // signature S
-      }, // signature
-    ]);
-
-    const receipt = await sendTransaction({
-      calls: [
-        {
-          contractAddress: ESCROW_ADDRESSES[chainId],
-          entrypoint: Entrypoint.CLAIM,
-          calldata: claimCallData,
-        },
-      ],
-    });
-
-    if (receipt.isSuccess()) {
-      showToast({type: 'success', title: 'Tip claimed successfully'});
-    } else {
-      let description = 'Please Try Again Later.';
-      if (receipt.isRejected()) {
-        description = receipt.transaction_failure_reason.error_message;
+    const connectedAccount = await waitConnection();
+    if (!connectedAccount) return;
+
+    const getNostrEvent = async (gasAmount: bigint) => {
+      const event = new NDKEvent(ndk);
+      event.kind = NDKKind.Text;
+      event.content = `claim: ${cairo.felt(depositId)},${cairo.felt(
+        connectedAccount.address,
+      )},${cairo.felt(STRK[chainId].address)},${gasAmount.toString()}`;
+      event.tags = [];
+
+      await event.sign();
+      return event.rawEvent();
+    };
+
+    const feeResult = await estimateClaim.mutateAsync(
+      await getNostrEvent(BigInt(0.0004 * Number(decimalsScale(18)))),
```

Comment:
Where this constant come from? Does it take into account gas token price fluctuations?

Reply:
It is only used during fee estimation, I didn't want it to be 0 during the estimation to correctly estimate the fees.